pkg/apis/serving/v1/revision_types.go

```diff
@@ -136,6 +136,13 @@ type RevisionStatus struct {
 	// may be empty if the image comes from a registry listed to skip resolution.
 	// +optional
 	ImageDigest string `json:"imageDigest,omitempty"`
```

Comment:
Mark this as deprecated, add a todo to remove in 2 releases?

Reply:
We don't remove stuff from the API, I think?

Reply:
🤷‍♂ we did before

Reply:
Yes while discussing during POC we decided like we will keep existing one to not to break the functionality

Reply:
I am not suggesting we remove it now. But in N knative releases (N=2, I think) this stops being of interest. So we if we rename it `deprecatedImageDigest` and a todo to remove it in future (0.16 I guess) it should be alright.

Reply:
Please note that this also forces clients that link against the API types to be updated. This immediately breaks any code (not only in two versions). E.g. see the kn client failure in https://storage.googleapis.com/knative-prow/pr-logs/pull/knative_client/801/pull-knative-client-unit-tests/1255135293493743616/build-log.txt:

```
Running tests with 'go test -v  -race -mod=vendor ./... '
# knative.dev/client/pkg/serving
pkg/serving/config_changes.go:305:24: baseRevision.Status.ImageDigest undefined (type "knative.dev/serving/pkg/apis/serving/v1".RevisionStatus has no field or method ImageDigest)
pkg/serving/config_changes.go:306:51: baseRevision.Status.ImageDigest undefined (type "knative.dev/serving/pkg/apis/serving/v1".RevisionStatus has no field or method ImageDigest)
# knative.dev/client/pkg/serving [knative.dev/client/pkg/serving.test]
pkg/serving/config_changes.go:305:24: baseRevision.Status.ImageDigest undefined (type "knative.dev/serving/pkg/apis/serving/v1".RevisionStatus has no field or method ImageDigest)
pkg/serving/config_changes.go:306:51: baseRevision.Status.ImageDigest undefined (type "knative.dev/serving/pkg/apis/serving/v1".RevisionStatus has no field or method ImageDigest)
pkg/serving/config_changes_test.go:128:17: revision.Status.ImageDigest undefined (type "knative.dev/serving/pkg/apis/serving/v1".RevisionStatus has no field or method ImageDigest)
```

This raises the question of what "deprecation" should mean. Only on the API (yaml) level or also on the code level ? Currently renaming fields to a `Deprecated` prefix is just a sledge-hammer for any client using a code dependency (also, this is not necessarily needed, or ?) and is, despite the naming, not a "deprecation" but a breaking change.